Card/src/main/java/com/braintreepayments/api/card/CardClient.kt

```diff
@@ -0,0 +1,141 @@
+package com.braintreepayments.api.card
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import com.braintreepayments.api.card.CardNonce.Companion.fromJSON
+import com.braintreepayments.api.core.ApiClient
+import com.braintreepayments.api.core.BraintreeClient
+import com.braintreepayments.api.core.BraintreeException
+import com.braintreepayments.api.core.Configuration
+import com.braintreepayments.api.core.GraphQLConstants
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Used to tokenize credit or debit cards using a [Card]. For more information see the
+ * [documentation](https://developer.paypal.com/braintree/docs/guides/credit-cards/overview)
+ */
+class CardClient {
+
+    private val braintreeClient: BraintreeClient
+    private val apiClient: ApiClient
+
+    /**
+     * Initializes a new [CardClient] instance
+     *
+     * @param context an Android Context
+     * @param authorization a Tokenization Key or Client Token used to authenticate
+     */
+    constructor(context: Context, authorization: String) : this(
+        BraintreeClient(
+            context,
+            authorization
+        )
+    )
+
+    @VisibleForTesting
+    internal constructor(braintreeClient: BraintreeClient) : this(
+        braintreeClient,
+        ApiClient(braintreeClient)
+    )
+
+    @VisibleForTesting
+    internal constructor(
+        braintreeClient: BraintreeClient,
+        apiClient: ApiClient
+    ) {
+        this.braintreeClient = braintreeClient
+        this.apiClient = apiClient
+    }
```

Comment:
We might be able to consolidate these internal constructors and move `braintreeClient` and `apiClient` to constructor properties.

See PayPalClient as an example: https://github.com/braintree/braintree_android/blob/f13626a796ae46b21fe41d62b0092c7f73d3c036/PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.kt#L22

Reply:
I Agree. We can finally leverage Kotlin default parameters 🙌 .